core/tests/protractor_desktop/coreEditorAndPlayerFeatures.js

```diff
@@ -33,6 +33,148 @@ var ExplorationEditorPage =
+    responseEditor = explorationEditorMainTab.getResponseEditor(0);
+    responseEditor.markAsCorrect();
+    explorationEditorMainTab.expectTickMarkIsDisplayed();
+    explorationEditorPage.saveChanges();
+  });
+
+  it('should allow selecting correct feedback from the response editor ' +
+     'during set the interaction', function() {
+    // Turn on correctness feedback first.
+    applyCommonSettings();
+
+    // Go to main tab to create interactions.
+    explorationEditorPage.navigateToMainTab();
+    explorationEditorMainTab.setStateName('First');
+    explorationEditorMainTab.setContent(forms.toRichText(
+      'Select the right option.'));
+
+    // Create interaction without close the add response modal. Set
+    // correctness in the modal.
+    explorationEditorMainTab.setInteractionWithoutCloseAddResponse('TextInput');
+    responseEditor = explorationEditorMainTab.getResponseEditor('pop');
+    responseEditor.markAsCorrect();
+
+    // Set the response for this interaction and close it.
+    explorationEditorMainTab.setResponse(
+      'TextInput', forms.toRichText('Correct!'),
+      'End', true, 'Equals', 'One');
+
+    explorationEditorMainTab.expectTickMarkIsDisplayed();
+    responseEditor = explorationEditorMainTab.getResponseEditor('default');
+    responseEditor.setFeedback(forms.toRichText('Wrong!'));
+    explorationEditorMainTab.moveToState('End');
+    explorationEditorMainTab.setInteraction('EndExploration');
+    explorationEditorPage.saveChanges();
+  });
+
+  it('should allow selecting correct feedback from the default response editor',
+    function() {
+      // Turn on correctness feedback first.
+      applyCommonSettings();
+
+      // Go back to main tab to create interactions.
+      explorationEditorPage.navigateToMainTab();
+      explorationEditorMainTab.setStateName('First');
+      explorationEditorMainTab.setContent(forms.toRichText(
+        'Select the right option.'));
+      explorationEditorMainTab.setInteraction('NumericInput');
+
+      // Set correctness in response editor.
+      responseEditor = explorationEditorMainTab.getResponseEditor('default');
+      responseEditor.markAsCorrect();
+      responseEditor.setFeedback(forms.toRichText('Correct!'));
+      responseEditor.setDestination('End', true, true);
+      explorationEditorMainTab.expectTickMarkIsDisplayed();
+
+      explorationEditorMainTab.moveToState('End');
+      explorationEditorMainTab.setInteraction('EndExploration');
+      explorationEditorPage.saveChanges();
+    });
+
+  afterEach(function() {
+    workflow.publishExploration();
+    libraryPage.get();
+    libraryPage.findExploration(explorationTitle);
+    libraryPage.playExploration(explorationTitle);
+    explorationPlayerPage.submitAnswer.apply(
+      null, correctOptions[currentExplorationIndex]);
+    explorationPlayerPage.expectCorrectFeedback();
+    explorationPlayerPage.clickThroughToNextCard();
+    explorationPlayerPage.expectExplorationToBeOver();
```

Comment:
I think the testing should be a part of `it` block, and beforeEach and afterEach should be for setup and cleanup respectively, so maybe we should move the tests (or the expected behavior of the test) in the `it block!)

@nithusha21 do you have any thoughts on this? Should we define a method on top to reduce duplicate code and use that method in `it` block? OR it's fine to have these test over here?

Reply:
@cuichenli I think a solution similar to applyCommonSettings would work here. Create a function and call them within the it blocks. Typically the complete test is present in the it() block, even if it is repetitive. In the afterEach block make sure that there are no console errors (You can see all other afterEach blocks follow that).

Reply:
I have moved some lines to `beforeEach` while leaving a function `enableMarkCorrectness` to just enable the option.